osu.Game/Skinning/RulesetResourcesSkin.cs

```diff
@@ -19,33 +22,65 @@ namespace osu.Game.Skinning
     /// </summary>
     public class RulesetResourcesSkin : ISkin
     {
-        private readonly TextureStore rulesetTextures;
-        private readonly ISampleStore rulesetSamples;
+        private readonly TextureStore? textures;
+        private readonly ISampleStore? samples;
 
         public RulesetResourcesSkin(Ruleset ruleset, GameHost host, AudioManager audio)
         {
-            IResourceStore<byte[]> rulesetResources = ruleset.CreateResourceStore();
+            IResourceStore<byte[]>? resources = ruleset.CreateResourceStore();
```

Comment:
Rather than all the internal null checks, may be worth passing just the non-null `IResourceStore` into this class (and not adding/instantiating it in the first place if it's null).